### PR TITLE
"In Graphics", including SVG image support

### PR DESCRIPTION
**Pull request by pickfifteen**
Still some work to do on this, all of which is technically optional:

- [ ] More grammar entries, especially in steps--this is most important. Currently there's few enough some steps will repeat on the same image frequently.
- [x] In-console preview needs some way to see what svg image will be generated; right now non-headline-tags are just ignored (same with existing image upload)
- [ ] Do we want to host the background image in the repo?
- [x] More background images that can be rotated
- [ ] Prettify the background image so it doesn't look so bad with white-bordered cards
- [ ] If someone is really ambitious, creating the background image from component parts in the SVG
- [ ] Come up with some way to fix the ugly white dots in the corners of the cards
- [ ] Handle two-lines in the title string better
- [ ] Look into using Beleren for title string?
- [ ] Double check all the real cards that they actually return an image from our image finder

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,9 +1,26 @@
...
 const numExamples = process.argv[2] || 30;
 const headlines = NewsEngine.generateHeadlines(numExamples);
 headlines.forEach(headline => {
 	console.log("\n * " + headline.text);
+
+	if(headline.tags && headline.tags.svg && headline.tags.svg.svgString) {
+		const outputfile = uuid() + '.png';
+		const outputPath = config.paths.tempDirectory + '/' + outputfile;
+
+		svg2png(new Buffer(headline.tags.svg.svgString))
```
Comment (pickfifteen) [inline src/index.js]: This is done async so all the images and the output for them pile up at the end of the run. Whether this is desirable or not is up for debate but it was the Easy(tm) way to do it so I did it for now.

Comment (jwoolley) [inline src/index.js]: Seems fine. We might want to put this behind an config flag that we can enable locally but can turn off for the bot, to keep the images from piling up.

**Comment by jwoolley** [comment]
I'm not on my best photoshop game tonight but I put together a few more backgrounds (which can wait, I just felt like making some)

File: bot.js [hunk 2]
```diff
@@ -40,6 +41,15 @@ function postCardImageTweet(status, cardName) {
 		.catch(e => logError('Failed to download image: ' + e));
 }
 
+function postSvgTweet(status, svgString, altText = "Rendered image") {
+	svg2png(new Buffer(svgString))
+		.catch(e => logError('Failed to create png: ' + e))
+		.then(data => twitter.uploadTwitterImageData(data.toString('base64')))
+		.catch(e => logError('Failed to upload image: ' + e))
+		.then(twitterImage => twitter.postImageTweet(twitterImage, altText, status))
+		.catch(e => logError('Failed to post tweet: ' + e));
```
Comment (mdiehr) [inline bot.js]: Can't we just catch once at the end and rely on the exception indicating what sort of error it was?

File: src/index.js (hunk 1 above)
Comment (mdiehr) [inline src/index.js]: I'll mess around with some config for this later. I'll probably set up a specific /tmp dir on the server for this app to use for everything.

BTW for gggolem I just let card images pile up in a folder forever (each card gets a deterministic filename and 
it uses this as a cache for gatherer results)

That obviously won't work for this use case, but the vps has plenty of space for now.

**Pull request merged**